ansible-playbooks/roles/deploy/tasks/main.yml

```diff
@@ -0,0 +1,186 @@
+---
+
+- name: Validate ENV variable
+  fail:
+    msg: "Required variable ENVhas not been provided"
+  when: (ENV is undefined) or (ENV| length == 0)
+
+- name: Validate none local deployment variables
+  fail:
+    msg: "Required variable {{ item }} has not been provided for {{ ENV }} deployment"
+  when:
+    - ENV != "local" 
+    - (vars[item] is undefined) or (vars[item]| length == 0)
+  loop:
+    - VAULT_ADDRESS
+    - VAULT_TOKEN
+  
+- name: Validate local deployment variables
+  fail:
+    msg: "Required variable {{ item }} has not been provided for local deployment"
+  when:
+    - ENV == "local" 
+    - (vars[item] is undefined) or (vars[item]| length == 0)  
+  loop:
+    - PULL_SECRET
+    - S3_BUCKET_NAME
+    - S3_ENDPOINT
+    - AWS_ACCESS_KEY
+    - AWS_SECRET_KEY
+    - SLACK_API_URL
+
+- name: Deploy Openshift GitOps
+  shell: |-
+    if [ {{ ENV }} != "local" ]; then
+      sed -i'' --expression 's@<VAULT_ADDRESS>@'{{ VAULT_ADDRESS }}'@g;s@<VAULT_TOKEN>@'{{ VAULT_TOKEN }}'@g' ./../../cluster-bootstrap/openshift-gitops/config/argocd.yaml
+    fi
+
+    printf "=====================Create Openshift Gitops Subscription...\n"
+    kubectl apply -k ./../../cluster-bootstrap/openshift-gitops/deploy
+
+    while [[ $(kubectl get ns openshift-gitops --no-headers --ignore-not-found | wc -l) -eq 0 ]]; 
+        do echo "Waiting for openshift-gitops namespace to be created" && sleep 30;
+    done
+
+    printf "=====================Waiting for Openshift Gitops start up and running...\n"
+    kubectl wait --for=condition=Ready pods --all -n openshift-gitops --timeout=5m
+    printf "=====================Openshift Gitops deploy successful!\n"
+
+    printf "=====================Create and config Openshift Gitops instance with Vault plugin configed...\n"
+    kubectl apply -k ./../../cluster-bootstrap/openshift-gitops/config
+
+    printf "=====================Waiting for configured Gitops repo server up and running...\n"
+    kubectl wait --for=condition=Ready pods --all -n openshift-gitops --timeout=5m
+    printf "=====================Openshift Gitops deploy successful!\n"
+
+    printf "=====================Use the following info to login Openshift Gitops web console:\n"
+    printf "Web console URL: https://"
+    kubectl get route openshift-gitops-server -n openshift-gitops -o 'jsonpath={.spec.host}'
+    printf "\n"
+    printf "# admin.username\n"
+    printf "admin\n"
+    oc extract secret/openshift-gitops-cluster -n openshift-gitops --to=-
+    sleep 10
+
+    while [[ $(kubectl get pods -n openshift-gitops --no-headers --ignore-not-found | awk '/openshift-gitops-repo-server/' | wc -l) -ne 1 ]]; 
+        do echo "Waiting for openshift-gitops-repo-server with vault plugin start up and running" && sleep 30;
+    done
+    printf "✓ Openshift Gitops deployed successful!"
+  register: result
```

Comment:
https://docs.ansible.com/ansible/latest/reference_appendices/faq.html#how-do-i-keep-secret-data-in-my-playbook since we have secrets here, we should avoid logging them

Reply:
Thanks! Very good point! In order easy to debug, I removed the log that include secret to resolve this issue.